docs: Add docs for GenSX Cloud (#589)
gensx-inc/gensx: website/docs/src/content/cli-reference/build.mdx

## Usage

```bash
gensx build <file> [options]
```

## Arguments

| Argument | Description                                                                                                    |
| -------- | -------------------------------------------------------------------------------------------------------------- |
| `<file>` | Workflow file to build (e.g., `src/workflows.tsx`). This should export a GenSX Workflow as the default export. |

## Options

| Option                  | Description                                     |
| ----------------------- | ----------------------------------------------- |
| `-o, --out-dir <dir>`   | Output directory for the built files.           |
| `-t, --tsconfig <file>` | Path to a custom TypeScript configuration file. |
| `-h, --help`            | Display help for the command.                   |

## Description

This command builds your GenSX workflow into an optimized bundle that can be deployed to GenSX Cloud. It:

- Transpiles TypeScript to JavaScript
- Bundles all dependencies
- Optimizes the code for production
- Prepares the workflow for deployment

> After building, the command outputs the path to the bundled file, which can be used with the `gensx deploy` command.

Reviewer: might be worth making this a link

## Examples

```bash
# Build a workflow with default options
gensx build src/workflows.tsx

# Build a workflow with a custom output directory
gensx build src/workflows.tsx --out-dir ./dist

# Build a workflow with a custom TypeScript configuration
gensx build src/workflows.tsx --tsconfig ./custom-tsconfig.json
```

## Notes

> - The build process requires that your workflow file exports a GenSX Workflow as the default export

Reviewer: I don't think this is true. I haven't seen us using default exports 
```suggestion
- The build process requires that your workflow file exports one ore more GenSX workflows
```
